src/components/src/main/java/org/apache/jmeter/visualizers/JSR223Listener.java

```diff
@@ -38,7 +38,7 @@
  */
 @TestElementMetadata(labelResource = "displayName")
 public class JSR223Listener extends JSR223TestElement
-    implements Cloneable, SampleListener, TestBean, Visualizer {
+    implements Cloneable, JMeterThreadBoundSampleListener, TestBean, Visualizer {
```

Comment:
Why is this change?

Is it really required?

Reply:
Hello @vlsi , 
Yes , as JSR223Listener must be executed within JMeterThread thread

Reply:
I do not follow that. Do you mean that every custom listener implementation must be updated and recompiled with JMeterThreadBoundSampleListener?

If that is the case, I don't think the PR is acceptable.

Reply:
No as I consider that if it does not implement the new interfaces, it should work as currently , so be executed in JMeterThread.
But I added this interface to identify explicitly Listeners that have this constraint vs others. To be frank I (re) discovered this thanks to tests failures when we introduced the queue and it took me some time to understand the problem.
So by clearly identifying this requirement of some listeners it is now obvious.

Reply:
>No as I consider that if it does not implement the new interfaces, it should work as currently , so be executed in JMeterThread.

Do you consider the new interface to be a part of the public API?
If that is the case, the users might implement that interface right after the version is released, then you can't add methods there anymore.

Reply:
Yes it is part of Plugin API.
We can add methods provided we are able to provide a "default" method

Reply:
Which methods do you expect there?

Reply:
As I said , for now I don't clearly see one.
But for example, for JMeterThreadBoundSampleListener, would it be more clear to provide a method like this one:
void sampleOccurred(SampleEvent e, JMeterContext context);

to indicating to implementer that he has access to JMeter thread context

Reply:
Do you intend to create JMeterThreadBoundTimer, JMeterThreadBoundConfig, and so on interfaces? That does not scale.

Reply:
Marker interfaces are from Java 1.4 era. They have no place in 2020.

I do not mean interfaces must not be used, however, I want to hear a strong justification that something needs to be a marker interface.

Modern tools (e.g. Gradle, JUnit5, Micronaut, Spring Framework, Spring Boot) never (?) use marker interfaces.